Wikipedia/Code/WMFSettingsViewController.m

```diff
@@ -387,50 +382,66 @@ - (void)tweakViewControllerPressedDone:(FBTweakViewController *)tweakViewControl
 
 - (nullable NSIndexPath *)indexPathForVisibleCellOfType:(WMFSettingsMenuItemType)type {
     return [self.tableView.indexPathsForVisibleRows wmf_match:^BOOL(NSIndexPath *indexPath) {
-        return ((WMFSettingsMenuItem *)[self.elementDataSource itemAtIndexPath:indexPath]).type == type;
+        return ([self.tableView cellForRowAtIndexPath:indexPath].tag == type);
     }];
 }
 
 - (void)reloadVisibleCellOfType:(WMFSettingsMenuItemType)type {
     NSIndexPath *indexPath = [self indexPathForVisibleCellOfType:type];
     if (indexPath) {
-        [self.elementDataSource replaceItemAtIndexPath:indexPath withItem:[WMFSettingsMenuItem itemForType:type]];
+        [self.tableView reloadRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationNone];
```

Comment:
I think `reloadVisibleCellOfType` will still need to update the datasource's respective row with a new `[WMFSettingsMenuItem itemForType:type]`, otherwise `reloadRowsAtIndexPaths` will just show what it was showing before. For example, I don't think login will show you that you've logged in otherwise. (Oh that reminds me - another thing to be sure to test after fix is that you can log in, delete the app, reinstall, go back to settings, and ensure it still shows you as being logged in.)